Framework/API/src/FileProperty.cpp

```diff
@@ -396,5 +401,55 @@ std::string FileProperty::convertExtension(const std::string &filepath) const {
   fullpath.setExtension(ext);
   return fullpath.toString();
 }
+
+/** Expand user variables in file path.
+ *  On UNIX, ~ is replaced by the user's home directory, if found.
+ *  On Windows, a combination of HOMEPATH and HOMEDRIVE will be used.
+ *  If the path contains no user variables, or expansion fails, the path is
+ *  returned unchanged, for errors to be dealt with by the calling function
+ *  @param filepath The path to expand
+ *  @return The expanded path
+ */
+std::string FileProperty::expandUser(const std::string &filepath) const {
+  auto start = filepath.begin();
+
+  if (*start != '~') // No user variables in filepath
+    return filepath;
+
+  // Position of the first slash after the variable
+  auto nextSlash = find_if(start, filepath.end(), &isSlash);
+
+  // UNIX-style home variable (ie ~/blah)
+  if (std::distance(start, nextSlash) == 1) {
+    char *home;
+
+    if (!(home = std::getenv("HOME")))
+      if (!(home = std::getenv("USERPROFILE")))
+        if (!(home = std::getenv("HOMEPATH")))
```

Comment:
It seems for Windows we'd want both the `~` only syntax, and the `~username` syntax that I didn't know existed.

I checked and the syntax `~user/blah` does _not_ work on my Windows 10. @josephframsay Can you check if it works on Windows 7? 
Maybe we should remove that second check, and just leave the `~`? Then the only difference we have is:
- for Linux we check `HOME`
- for Windows we check `HOME` and `USERPROFILE`

As I said in the previous comment, `HOMEPATH` did not seem to include the drive letter